roles/openshift_facts/library/openshift_facts.py

```diff
@@ -997,6 +1002,23 @@ def merge_facts(orig, new, additive_facts_to_overwrite, protected_facts_to_overw
     return facts
 
 
+def merge_provider_facts(facts):
+    if 'provider' not in facts:
+        return facts
+    if 'master' in facts:
+        for arg in ('api_server_args', 'controller_args'):
+            facts['master'][arg] = merge_facts(
+                facts['provider'].get(arg, {}),
+                facts['master'].get(arg, {}),
+                [])
```

Comment:
There is now a fourth arg to merge_facts (protected_facts_to_overwrite), you'll need to add an additional '[]' to the argument list here.